docs/install.md

```diff
@@ -106,3 +107,34 @@ That way you have control over when a breaking change affects you.
     # │        │        │        │        │        │        │        │        │        │        │
     # (1, 0)───(1, 1)───(1, 2)───(1, 3)───(1, 4)───(1, 5)───(1, 6)───(1, 7)───(1, 8)───(1, 9)───(1, 10)
     ```
+
+
+### Installing on Docker
+
+This will create a Docker image containing Cirq's source code. It will isolate a Cirq installation from the rest of the system.
+
+0. [Install Docker](https://docs.docker.com/install/#supported-platforms) on your host sytem.
+
+1. Build the docker image for your system:
+    ```bash
+    git clone https://github.com/quantumlib/Cirq
```

Comment:
Maybe I've fundamentally misunderstood how people use docker, but it seems very odd for the user docker setup to involve cloning the source code? Shouldn't it be more along the lines of "here is a docker file that includes cirq==0.4.* as a dependency, and here is the dependency line you can paste into a docker file to include cirq"?

Reply:
That's true. All that's needed is the Dockerfile for manual building. We could host the Dockerfile independently so users can download it to avoid cloning the whole repo just to access the Dockerfile or; based on my comments from https://github.com/quantumlib/Cirq/pull/957, create a Docker Hub account so that the users don't have to clone the source or download the Dockerfile to build and use the image. They can just `docker pull quantumlib/cirq` and be ready to go. The latter is standard in the Docker community.

Reply:
We can also host the image independently.

Reply:
I have updated the development Dockerfile to use the requirements files, and the user Dockerfile now just pulls it it via cirq package. I suggest we do the prebuilt image in a different PR.